src/CollisionSolverEntities.s

```diff
@@ -0,0 +1,55 @@
+.ENUM $C000
+	collidingEntity DW
+.ENDE
+
+; //// Collision Solver for Entities other than Isaac \\\\
+collisionSolverIsaac :
+	push c
+	push d
+	push e
+	
+	ld c, hl ;get element's size and flags
+	ld hl, (collidingEntity)
+	inc hl
+	inc hl ;hl leads to entity's speed
+	
+	;/// Blocking Collision \\\
+	bit 4, c
+	jp z, noEBlockCollision
+
+	;// Touch Horizontally \\
+	bit 1, b
+	jr z, noEHorizontalCollision
+	ld a, (hl)
+	and %11110000
+	srl a
+	srl a
+	srl a
+	srl a

```

Comment:
you can use SWAP in this case (4 bitshifts to the right : SWAP the nibbles and AND with 0x0F)